ably/transport/defaults.py

```diff
@@ -47,12 +47,17 @@ def get_port(options):
 
     @staticmethod
     def get_fallback_rest_hosts(options):
-        if options.rest_host:
+        if (options.rest_host or options.environment or \
```

Comment:
This logic is wrong and could be addressed more easily with something like:

```
using_custom_host = options.rest_host or options.environment.lower() != 'production'
if options.fallback_hosts:
  fallback_hosts_copy = list(options.fallback_hosts)
elif using_custom_host && options.fallback_hosts_use_default:
  fallback_hosts_copy = list(Defaults.fallback_hosts)
elif !using_custom_host:
  fallback_hosts_copy = list(Defaults.fallback_hosts)
else:
  return []
end
...
```

Please recheck the specs to work out what the logic is